src/napari_clusters_plotter/_new_plotter_widget.py

```diff
@@ -522,6 +526,30 @@ def _on_update_layer_selection(
                     f"Layer {layer.name} does not have events.features or events.properties"
                 )
 
+            if self._is_selectable(layer):
+                event = self._get_selection_event(layer)
+                self._update_layer_selected_data_feature(layer)
+                event.connect(
+                    lambda e: self._update_layer_selected_data_feature(layer)
+                )
+
+        self._update_feature_selection(None)
+
+    def _update_layer_selected_data_feature(
+        self, layer: napari.layers.Layer
+    ) -> None:
+        """
+        Update the layer selected_data to feature.
+        """
+        selected_data = self._get_selected_objects(layer)
+        cluster = np.zeros(len(layer.features))
+        cluster[list(selected_data)] = 1
+        # set categorical to be selectable in "Hue" dropdown
+        layer.features["SELECTED_DATA_LAYER_CLUSTER_ID"] = pd.Categorical(
+            cluster
+        )
+        self.plot_needs_update.emit()
```

Comment:
Not sure - does it make sense to iterate over all the layers and collect the selected objects inside this function? That way, you could save yourself the `lambda` callback connection above. Somethhing like this:

```python
for layer in self.layers:
  selected_data = self._get_selected_objects(layer)
  cluster = np.zeros(len(layer.features))
  cluster[list(selected_data)] = 1
  # set categorical to be selectable in "Hue" dropdown
  layer.features["SELECTED_DATA_LAYER_CLUSTER_ID"] = pd.Categorical(cluster)
```